src/components/NavBar/ChainSwitcher.tsx

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useWeb3React } from '@web3-react/core'
+import { getChainInfo } from 'constants/chainInfo'
+import { SupportedChainId } from 'constants/chains'
+import { useOnClickOutside } from 'hooks/useOnClickOutside'
+import useSelectChain from 'hooks/useSelectChain'
+import useSyncChainQuery from 'hooks/useSyncChainQuery'
+import { Box } from 'nft/components/Box'
+import { Column, Row } from 'nft/components/Flex'
+import { NewChevronDownIcon, NewChevronUpIcon } from 'nft/components/icons'
+import { CheckMarkIcon } from 'nft/components/icons'
+import { subhead } from 'nft/css/common.css'
+import { ReactNode, useReducer, useRef } from 'react'
+import { isChainAllowed } from 'utils/switchChain'
+
+import * as styles from './ChainSwitcher.css'
+import { NavDropdown } from './NavDropdown'
+
+const ChainRow = ({
+  targetChain,
+  onSelectChain,
+}: {
+  targetChain: SupportedChainId
+  onSelectChain: (targetChain: number) => void
+}) => {
+  const { provider, chainId } = useWeb3React()
+  if (!provider || !chainId) {
+    return null
+  }
+  const active = chainId === targetChain
+  const { label, logoUrl } = getChainInfo(targetChain)
+
+  return (
+    <Row
+      as="button"
+      background={active ? 'lightGrayContainer' : 'none'}
+      className={`${styles.ChainSwitcherRow} ${subhead}`}
+      onClick={() => onSelectChain(targetChain)}
+    >
+      <ChainDetails>
+        <img src={logoUrl} alt={label} className={styles.Icon} />
+        {label}
+      </ChainDetails>
+      {active && <CheckMarkIcon width={20} height={20} />}
+    </Row>
+  )
+}
+
+const ChainDetails = ({ children }: { children: ReactNode }) => <Row>{children}</Row>
+
+const NETWORK_SELECTOR_CHAINS = [
+  SupportedChainId.MAINNET,
+  SupportedChainId.POLYGON,
+  SupportedChainId.OPTIMISM,
+  SupportedChainId.ARBITRUM_ONE,
+]
+
+interface ChainSwitcherProps {
+  isMobile?: boolean
+}
+
+export const ChainSwitcher = ({ isMobile }: ChainSwitcherProps) => {
+  const { chainId, connector } = useWeb3React()
+  const [isOpen, toggleOpen] = useReducer((s) => !s, false)
+
+  const ref = useRef<HTMLDivElement>(null)
+  useOnClickOutside(ref, isOpen ? toggleOpen : undefined)
+
+  const info = chainId ? getChainInfo(chainId) : undefined
+
+  const selectChain = useSelectChain()
+  useSyncChainQuery()
+
+  // TODO: need to handle unsupported chain
```

Comment:
can we handle this?

Reply:
Handling unsupported chains in not part of the current product spec. Additionally this is the same logic as is in `src/components/Header/NetworkSelector.tsx` ln297. I'll remove the comment since it just causes confusion

Reply:
ah sounds good